photoshop/api/_core.py

```diff
@@ -1,179 +1,118 @@
 """This class provides all photoshop API core functions."""
 # Import built-in modules
+from contextlib import suppress
+from functools import cached_property
+from logging import CRITICAL
+from logging import DEBUG
+from logging import Logger
+from logging import getLogger
 import os
 import platform
 from typing import Any
 from typing import List
+from typing import Optional
 import winreg
 
 # Import third-party modules
 from comtypes.client import CreateObject
-from comtypes.client import GetActiveObject
 from comtypes.client.dynamic import _Dispatch as FullyDynamicDispatch
+from comtypes.client.lazybind import Dispatch
 
 # Import local modules
-from photoshop.api import constants
+from photoshop.api.constants import PHOTOSHOP_VERSION_MAPPINGS
 from photoshop.api.errors import PhotoshopPythonAPIError
 
 
 class Photoshop:
     """Core API for all photoshop objects."""
 
     _root = "Photoshop"
-    REG_PATH = "SOFTWARE\\Adobe\\Photoshop"
-    _object_name = "Application"
-    object_name: str = None
-
-    def __init__(self, ps_version: str = None, parent: Any = None):
-        self._program_name = None
-        self._has_parent = False
-        version_mappings = constants.PHOTOSHOP_VERSION_MAPPINGS
-        self.photoshop_version = os.getenv("PS_VERSION", ps_version)
-        if self.photoshop_version:
-            # Store current photoshop version.
-            os.environ["PS_VERSION"] = self.photoshop_version
-        self.app_id = version_mappings.get(self.photoshop_version, "0")
-        try:
-            self.app = self.instance_app(self.app_id)
-        except OSError:
-            # get photoshop from registration.
-            try:
-                self.app = self.instance_app(self._get_program_id())
-            except OSError:
-                raise PhotoshopPythonAPIError(
-                    "Please check if you have Photoshop installed correctly.",
+    _reg_path = "SOFTWARE\\Adobe\\Photoshop"
+    object_name: str = "Application"
```

Comment:
Deprecated `_object_name` since it just replaces object_name if object_name is None, makes more sense to just have object_name inherited as Application if its not set by the child class.